### PR TITLE
docs: fix minor layout issues and allow expanding nav sections, update docs around ssr and component import options

### PR DESCRIPTION
**Pull request by borisdiakur**
# Description

See above ☝️

## Type of change

Please delete options that are not relevant.

- [x] Improves docs page

# How Has This Been Tested?

- [x] tested manually

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: src/docs/components/docs-topbar/docs-topbar.css [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
 
 .docs-topbar__logo {
   display: block;
+  margin-left: calc(-1 * var(--ld-sp-6));
+}
+
+.docs-topbar__links {
+  margin-right: 0;
+  width: var(--docs-toc-width);
+  padding-left: calc(var(--ld-sp-24));
```
Comment (renet) [inline src/docs/components/docs-topbar/docs-topbar.css:18]:
```suggestion
  padding-left: var(--ld-sp-24);
```

I also think this looks a little odd. Can you maybe make a little design review with Nic, first?

File: src/docs/layouts/docs-layout/docs-layout.css [hunk 2]
```diff
@@ -1,6 +1,9 @@
 body {
   overscroll-behavior-x: none;
   -webkit-text-size-adjust: none;
+  --docs-max-width: 90rem;
+  --docs-nav-width: 17rem;
+  --docs-toc-width: 17rem;
```
Comment (renet) [inline src/docs/layouts/docs-layout/docs-layout.css:6]: What does "toc" stand for, anyway?

File: src/docs/components/docs-topbar/docs-topbar.css [hunk 3]
```diff
@@ -1,19 +1,34 @@
...
+
+  &::part(container) {
+    margin: 0 var(--ld-sp-4);
+    padding: 0 var(--ld-sp-24);
+  }
 }
 
 .docs-topbar__logo {
   display: block;
+  margin-left: calc(-1 * var(--ld-sp-6));
```
Comment (renet) [inline src/docs/components/docs-topbar/docs-topbar.css:12]: In the breakpoint where the navigation is permanently visible, the logo does now no longer align with the navigation (icons). Same goes for the icons/links on the right side - maybe we should add another breakpoint with increased `.docs-topbar::part(container)` padding for the in-between breakpoint?

**Review by renet — CHANGES_REQUESTED** [review]

File: src/docs/layouts/docs-layout/docs-layout.css (hunk 2 above)
Comment (borisdiakur) [reply src/docs/layouts/docs-layout/docs-layout.css:6]: table of contents

File: src/docs/components/docs-nav/docs-nav.css [hunk 4]
```diff
@@ -68,24 +65,40 @@
...
   .docs-nav__summary--introduction,
+  .docs-nav__summary--introduction + ul .docs-nav__summary,
+  .docs-nav__summary--globals,
+  .docs-nav__summary--globals + ul .docs-nav__summary,
+  .docs-nav__summary--components,
+  .docs-nav__summary--components + ul .docs-nav__summary {
+    .docs-nav__summary-toggle {
+      display: block;
+    }
+  }
```
Comment (renet) [inline src/docs/components/docs-nav/docs-nav.css:76]:
```suggestion
  .docs-nav__summary {
    .docs-nav__summary-toggle {
      display: block;
    }
  }
```

Comment (renet) [reply src/docs/components/docs-nav/docs-nav.css:76]: Alternatively, I would move the `display: block;` down and have it 3x the same - still less css written. ^^

File: src/docs/components/docs-nav/docs-nav.css [hunk 5]
```diff
@@ -195,20 +210,7 @@
...
-      background-repeat: no-repeat;
-      background-position: center;
-      width: 0.8rem;
-      height: 1rem;
-      display: inline-flex;
-      align-items: center;
-      justify-content: center;
-      pointer-events: none;
-    }
+    pointer-events: all;
```
Comment (renet) [inline src/docs/components/docs-nav/docs-nav.css:213]: This applies to svg only, according to MDN. Does this work with inheritance?

File: src/docs/components/docs-topbar/docs-topbar.css [hunk 6]
```diff
@@ -3,23 +3,32 @@
 
   &::part(container) {
     margin: 0 var(--ld-sp-4);
-    padding: 0 var(--ld-sp-24);
+    padding: 0 var(--ld-sp-32);
```
Comment (renet) [inline src/docs/components/docs-topbar/docs-topbar.css:6]: Wouldn't it be easier to simply add the breakpoint media queries here, so you don't have to work with negative margins?

File: src/docs/components/docs-topbar/docs-topbar.tsx [hunk 7]
```diff
@@ -8,12 +8,14 @@ import { Component, h, getAssetPath } from '@stencil/core'
...
 export class DocsTopbar {
+  private base = document.querySelector('base').href
+
   render() {
     return (
       <ld-header
         class="docs-topbar ld-theme-bubblegum"
         logo-title="Home"
-        logo-url="/"
+        logo-url={this.base}
```
Comment (renet) [inline src/docs/components/docs-topbar/docs-topbar.tsx:18]: Shouldn't the absolute `logo-url` (actually a href attribute) depend on the base value in the head, anyway?

File: src/docs/pages/server-side-rendering.md [hunk 8]
```diff
@@ -16,20 +16,26 @@ Custom elements should be defined on client-side only. There are a couple of thi
...
 ```
 
 You can also import the components you use on client side only, if you want to leverage code splitting and lazy loading:
 
 ```js
 if (typeof window !== 'undefined') {
-  import('@emdgroup-liquid/liquid/dist/custom-elements').then((liquid) => {
-    const { LdButton } = liquid
-    customElements.define('ld-button', LdButton)
+  const { setAssetPath } = await import('@emdgroup-liquid/liquid/dist/components')
```
Comment (renet) [inline src/docs/pages/server-side-rendering.md:30]:
```suggestion
```

File: src/docs/pages/server-side-rendering.md [hunk 9]
```diff
@@ -16,20 +16,26 @@ Custom elements should be defined on client-side only. There are a couple of thi
...
+  const { setAssetPath } = await import('@emdgroup-liquid/liquid/dist/components')
+    const modules = await Promise.all([
+      import('@emdgroup-liquid/liquid/dist/components/ld-button'),
+      import('@emdgroup-liquid/liquid/dist/components/ld-checkbox'),
+    ])
```
Comment (renet) [inline src/docs/pages/server-side-rendering.md:35]:
```suggestion
  const modules = await Promise.all([
    import('@emdgroup-liquid/liquid/dist/components/ld-button'),
    import('@emdgroup-liquid/liquid/dist/components/ld-checkbox'),
  ])
```

**Review by renet — APPROVED** [review]

File: src/docs/components/docs-nav/docs-nav.css (hunk 5 above)
Comment (borisdiakur) [reply src/docs/components/docs-nav/docs-nav.css:208]: Yes it works well. Have been using this technique for years 🤓

File: src/docs/components/docs-topbar/docs-topbar.css (hunk 6 above)
Comment (borisdiakur) [reply src/docs/components/docs-topbar/docs-topbar.css:6]: I don't know. I'll leave it as it is.

File: src/docs/components/docs-topbar/docs-topbar.tsx (hunk 7 above)
Comment (borisdiakur) [reply src/docs/components/docs-topbar/docs-topbar.tsx:18]: It seems it doesn't. Currently setting an absolute URL with `/` results in a navigation to https://emdgroup-liquid.github.io/ which shows a 404.

**Pull request merged**